pkg/i2gw/providers/common/converter.go

```diff
@@ -337,9 +341,35 @@ func (rg *ingressRuleGroup) toHTTPRoute(servicePorts map[types.NamespacedName]ma
 		errors = append(errors, errs...)
 		hrRule.BackendRefs = backendRefs
 
+		// Add source ingress information for this specific rule
+		if hrRule.Filters == nil {
+			hrRule.Filters = []gatewayv1.HTTPRouteFilter{}
+		}
+
+		for _, p := range paths {
+			ruleSourceIngresses[p.sourceIngress] = append(ruleSourceIngresses[p.sourceIngress], len(httpRoute.Spec.Rules))
+		}
+
 		httpRoute.Spec.Rules = append(httpRoute.Spec.Rules, hrRule)
 	}
 
+	// Initialize annotations map if needed
+	if httpRoute.Annotations == nil {
+		httpRoute.Annotations = make(map[string]string)
+	}
+
+	// Build annotation mapping source ingress to rule indices
+	var annotationParts []string
```

Comment:
When initializing a slice, it's best to set its size if you know it beforehand. Saves on having to reallocate whenever it grows for each append.

`annotationParts := make([]string, 0, len(ruleSourceIngresses))`